### PR TITLE
Adds drafts idea to events

### PR DESCRIPTION
**Pull request by siruguri**
Pivotal: https://www.pivotaltracker.com/story/show/83605984

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -48,19 +48,29 @@ def edit
...
-      else
-        EventMailer.unpublished_event(@event).deliver_now
-        EventMailer.event_pending_approval(@event).deliver_now
-      end
 
-      if @event.published
-        redirect_to @event, notice: 'Event was successfully created.'
-      else
+      case @event.current_state
+      when nil # No state set => it's now pending approval
```
Comment (lilliealbert) [inline app/controllers/events_controller.rb]: I wonder if having an explicit state for pending approval would be more clear than relying on a lack of one?

Comment (siruguri) [inline app/controllers/events_controller.rb]: Agreed - brain fart on my part; I too don't like having non-semantic data
values. I'll wait a bit for more comments, and correct that in a PR update.

**Comment by siruguri** [comment]
I also realized that my code leaves the `draft_saved` attribute incorrectly set to `true` for previously-drafted events. I'll fix that as well - so pls ignore this PR for now.

**Comment by tjgrathwell** [comment]
Looks good, two little things:
* The `.travis.yml` file needs to be updated with the new ruby version for the TravisCI build to run
* The `draft_saved` migration date is in April but there were other migrations created and ran in May. The effect is such that if you run `rake db:migrate` against an existing DB it dumps an updated schema where the `draft_saved` column is moved last. Maybe rename the migration file so it takes place last?

**Comment by siruguri** [comment]
@travis - thanks.

* I was looking for that file - forgot to add a `-a` to my `ls`, thought I'd see a `travis.yml` without a dot.
* Hmm yeah, I take it you mean that on an existing DB, the `draft_saved` migration wouldn't be run at all? I'll just `git mv` it to a newer date.

**Comment by tjgrathwell** [comment]
Can you squash the amendments into the previous commit?

**Comment by tjgrathwell** [comment]
...meaning probably squash all three commits together; ideally every commit in the timeline is one that would have passed on TravisCI, and right now separating the ruby upgrade from the .travis.yml change makes that not true. Alternatively you could just merge the .travis.yml fix into the initial ruby bump commit, whichever feels reasonable to you.

**Comment by siruguri** [comment]
I'll squash all, np.

Typed on a mobile device.

**Comment by siruguri** [comment]
Ok, squashed together.

**Comment by tjgrathwell** [comment]
Alright, it's merged and live! Thanks a lot for this!

**Pull request merged**